[const_fold] Set requires_grad based on the folded tensor; add device_for_folding option

jfix71 opened the pull request:
Summary: att

Test Plan: Added unit test coverage for tensor_meta part.

Reviewed By: wushirong

Differential Revision: D36975932

dborkovic reviewed (approved):
Other than checking that the results are equal in line 670, do we need to check that the graph structure has changed in any particular fashion?

jfix71 committed:
[const_fold] Set requires_grad based on the folded tensor; add device_for_folding option (#90)

Summary:
X-link: https://github.com/pytorch/fx2trt/pull/90

Pull Request resolved: https://github.com/pytorch/pytorch/pull/79067

att

Test Plan: Added unit test coverage for tensor_meta part.

Reviewed By: yinghai

Differential Revision: D36975932

fbshipit-source-id: 228bc2d501f411b6fc2aeeb7efe5140a631697ea
